src/supertux/level_parser.cpp

```diff
@@ -35,6 +36,49 @@ LevelParser::from_file(const std::string& filename)
   return level;
 }
 
+std::unique_ptr<Level>
+LevelParser::from_nothing(const std::string& basedir)
+{
+  std::unique_ptr<Level> level(new Level);
+  LevelParser parser(*level);
+
+  // Find a free level filename
+  std::string level_file;
+  int num = 0;
+  do {
+    num++;
+    level_file = basedir + "/level" + std::to_string(num) + ".stl";
+  } while ( PHYSFS_exists(level_file.c_str()) );
+  std::string level_name = "Level " + std::to_string(num);
+  level_file = "level" + std::to_string(num) + ".stl";
+
+  parser.create(level_file, level_name, false);
+  return level;
+}
+
+std::unique_ptr<Level>
+LevelParser::from_nothing_worldmap(const std::string& basedir, const std::string& name)
+{
+  std::unique_ptr<Level> level(new Level);
+  LevelParser parser(*level);
+
+  // Find a free level filename
+  std::string level_file = basedir + "/worldmap.stwm";
+  if (PHYSFS_exists(level_file.c_str())) {
+    int num = 0;
+    do {
+      num++;
+      level_file = basedir + "/worldmap" + std::to_string(num) + ".stl";
+    } while ( PHYSFS_exists(level_file.c_str()) );
+    level_file = "worldmap" + std::to_string(num) + ".stl";
```

Comment:
You should be aware that we don't parse worldmap<num>.stl at all. What's the point in saving a worldmap if it can't be used? If anything, this should be .stwm.